### PR TITLE
fix: undefined check for asset

### PR DESCRIPTION
**Pull request by milljoniaer**
### Description

This PR fixes the build error:
image

Somehow this error only accurs with the images of the new blog article by eric and not the other ones. But this fixes the build error. Also the main build fails currently because of this.

### CODE REVIEW

File: gatsby/config-options/gatsby-plugin-feed-options.tsx [hunk 1]
```diff
@@ -106,6 +106,11 @@ export const rssRenderOptions: Partial<Options> = {
       }
     },
     [BLOCKS.EMBEDDED_ASSET]: (node) => {
+      if (!node.data.target.file) {
+        console.log(JSON.stringify(node));
```
Comment (georgiee) [inline gatsby/config-options/gatsby-plugin-feed-options.tsx:110]: well maybe that's a little bit too debugging? Could you log a message and maybe mention some other details.
Or alternative to get things going: Remove the console .log
Or alternative 2: I know that you're working on this topic anyway so let it be  😅

**Review by georgiee — APPROVED** [review]
Thanks for the quick fix.

File: gatsby/config-options/gatsby-plugin-feed-options.tsx (hunk 1 above)
Comment (milljoniaer) [reply gatsby/config-options/gatsby-plugin-feed-options.tsx:110]: Thank you for your review, I added a fix for the original problem which should be a better way of fixing the build

File: gatsby/config-options/gatsby-plugin-feed-options.tsx [hunk 2]
```diff
@@ -197,26 +231,8 @@ const feedOptions = {
           if (introRaw.content)
             rawContent.content = [...introRaw.content, ...copyRaw.content];
 
-          const contentWithEmbeddedEntries = rawContent.content.map(
```
Comment (georgiee) [inline gatsby/config-options/gatsby-plugin-feed-options.tsx:200]: I see that you refactored this part into a dedicated function. That makes it difficult to find the actual change. What did you fix aside from the refactoring? What was the issue? Can you highlight it on the counter part if this removal

Comment (milljoniaer) [reply gatsby/config-options/gatsby-plugin-feed-options.tsx:200]: On all the other types than images, (so paragraphs, lists and so on) i did not only return the content as it was in there previous version, but checked if it has children (content.content), and then call the function recursively on those children.

In that way we get all images, even if they are very cascadenated.

Comment (georgiee) [reply gatsby/config-options/gatsby-plugin-feed-options.tsx:200]: nice, thanks for the clarification!

File: gatsby/config-options/gatsby-plugin-feed-options.tsx [hunk 3]
```diff
@@ -152,6 +157,35 @@ export const rssRenderOptions: Partial<Options> = {
...
+  // Skip everything but entry and asset blocks
+  if (
+    content.nodeType !== 'embedded-entry-block' &&
+    content.nodeType !== 'embedded-asset-block'
+  ) {
+    if (content.content) {
+      return {
+        ...content,
+        content: content.content.map((el) =>
+          addImageDataToLinks({ content: el, node }),
```
Comment (georgiee) [inline gatsby/config-options/gatsby-plugin-feed-options.tsx:170]: power to the recursion. thanks for handling this case so easily 👍

**Review by georgiee — APPROVED** [review]
here we go, my second approval 🙌 Great how you solved it.
I did not run any code and just checked the code on the surface combined with your given feedback. Thank you.

**Pull request merged**